### PR TITLE
FE: Monitoring Policies - component unit tests

### PR DESCRIPTION
**Pull request by faizanjanjua**
## Description
Get the Monitoring Policies component unit tests working again (/tests/MonitoringPolicies/monitoringPolicies.test.ts).

## Jira link(s)
- https://opennms.atlassian.net/browse/LOK-2557

## Checklist
* [ ] Follows Lōkahi's [development guidelines.](https://github.com/OpenNMS-Cloud/lokahi/wiki/Development-Guidelines)
* [ ] Appropriate reviewer(s) have been selected.
* [ ] Jira issue(s) have been updated to "In Review".
* [ ] Includes [appropriate tests.](https://github.com/OpenNMS-Cloud/lokahi/wiki/Test-Strategy)
* [ ] Documentation has been updated as necessary.
* [ ] Notify devops of changes to the Charts.
* [ ] Notify documentation team of any changes to names of screens or features (affects URLs).

### CODE REVIEW

**Review by Shahbaz-dataq — COMMENTED** [review]
Just a small change

File: ui/tests/MonitoringPolicies/monitoringPolicies.test.ts [hunk 1]
```diff
@@ -83,95 +16,31 @@ const wrapper = mount({
...
-    const store = useMonitoringPoliciesStore()
-    const newPolicyBtn = wrapper.get('[data-test="new-policy-btn"]')
-
-    expect(store.selectedPolicy).toStrictEqual(defaultPolicy)
+    const store =useMonitoringPoliciesStore()
+    const newPolicyBtn = wrapper.get('[data-test="create-policy-btn"]')
+    expect(store.selectedPolicy).toBe(undefined)
     await newPolicyBtn.trigger('click')
     expect(store.displayPolicyForm).toHaveBeenCalledTimes(1)
     expect(store.selectedPolicy).toBeTruthy()
```
Comment (Shahbaz-dataq) [inline ui/tests/MonitoringPolicies/monitoringPolicies.test.ts:38]: Create a `defaultPolicy` with wrapper just like in `monitoringPolicyStore` and check it here for `toBeStrickEqual`

Comment (faizanjanjua) [reply ui/tests/MonitoringPolicies/monitoringPolicies.test.ts]: done

**Review by Shahbaz-dataq — APPROVED** [review]
Looks good.

File: ui/src/containers/MonitoringPolicies.vue [hunk 2]
```diff
@@ -13,12 +13,14 @@
         <section class="feather-row">
           <div :class="displayDetails ? 'feather-col-8' : 'feather-col-12'">
             <MonitoringPoliciesTable
+            data-test="MonitoringPoliciesTable"
```
Comment (synqotik) [inline ui/src/containers/MonitoringPolicies.vue:16]: Indent 2 spaces

File: ui/src/containers/MonitoringPolicies.vue [hunk 3]
```diff
@@ -13,12 +13,14 @@
...
               @policy-selected="displayDetails = true"
               :refreshMonitoringPolicies = "reFresh"
             />
           </div>
           <div v-if="displayDetails" class="feather-col-4">
             <MonitoringPoliciesDetailPanel
+            data-test="MonitoringPoliciesDetailPanel"
```
Comment (synqotik) [inline ui/src/containers/MonitoringPolicies.vue:23]: Indent 2 spaces

File: ui/tests/MonitoringPolicies/monitoringPolicies.test.ts [hunk 4]
```diff
@@ -1,77 +1,24 @@
...
-const defaultRule: PolicyRule = {
-  id: 0,
-  name: '',
-  componentType: ManagedObjectType.Node,
-  detectionMethod: DetectionMethod.Event,
-  eventType: EventType.SnmpTrap,
-  alertConditions: [],
-  vendor: 'generic'
+const mockMonitoringPolicy: Policy = {
+
```
Comment (synqotik) [inline ui/tests/MonitoringPolicies/monitoringPolicies.test.ts:11]: Remove blank line, here and at end of this definition

File: ui/tests/MonitoringPolicies/monitoringPolicies.test.ts [hunk 5]
```diff
@@ -83,95 +30,31 @@ const wrapper = mount({
...
-
-  test('The store populates with a selected policy when "New Policy" is clicked.', async () => {
-    const store = useMonitoringPoliciesStore()
-    const newPolicyBtn = wrapper.get('[data-test="new-policy-btn"]')
-
-    expect(store.selectedPolicy).toStrictEqual(defaultPolicy)
-    await newPolicyBtn.trigger('click')
-    expect(store.displayPolicyForm).toHaveBeenCalledTimes(1)
-    expect(store.selectedPolicy).toBeTruthy()
+  test('Test if the create-policy-btn exist' , () => {
```
Comment (synqotik) [inline ui/tests/MonitoringPolicies/monitoringPolicies.test.ts:37]: `exist` -> `exists`

File: ui/tests/MonitoringPolicies/monitoringPolicies.test.ts [hunk 6]
```diff
@@ -83,95 +30,31 @@ const wrapper = mount({
...
-  test('The store populates with a selected rule when "New Rule" is clicked.', async () => {
-    const store = useMonitoringPoliciesStore()
-    const newRuleBtn = wrapper.get('[data-test="new-rule-btn"]')
-
-    expect(store.selectedRule).toStrictEqual(defaultRule)
-    await newRuleBtn.trigger('click')
-    expect(store.displayRuleForm).toHaveBeenCalledTimes(1)
-    // FIXME: This test broke after displayRuleForm was made async
-    // expect(store.selectedRule).toBeTruthy()
+  test('Test if the MonitoringPoliciesTable component to exist' , () => {
```
Comment (synqotik) [inline ui/tests/MonitoringPolicies/monitoringPolicies.test.ts:40]: `to exist` -> `exists`

File: ui/tests/MonitoringPolicies/monitoringPolicies.test.ts [hunk 7]
```diff
@@ -83,95 +30,31 @@ const wrapper = mount({
...
-
-    // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
-    expect(store.selectedPolicy!.rules?.length).toBe(0)
-    await wrapper.get('[data-test="rule-name-input"] .feather-input').setValue('Rule1')
-    await saveRuleBtn.trigger('click')
-
-    expect(store.saveRule).toHaveBeenCalledTimes(1)
-    // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
-    expect(store.selectedPolicy!.rules?.length).toBe(1)
+  test('Test if the MonitoringPoliciesDetailPanel component to exist' , () => {
```
Comment (synqotik) [inline ui/tests/MonitoringPolicies/monitoringPolicies.test.ts:43]: `to exist` -> `exists`

**Review by synqotik — CHANGES_REQUESTED** [review]
Looks good, just a few small changes.

File: ui/tests/MonitoringPolicies/monitoringPolicies.test.ts (hunk 5 above)
Comment (faizanjanjua) [reply ui/tests/MonitoringPolicies/monitoringPolicies.test.ts]: done

**Review by synqotik — APPROVED** [review]

**Pull request merged**